modules/javafx.graphics/src/test/java/test/javafx/scene/FocusTest.java

```diff
@@ -1128,4 +1128,35 @@ class N extends Group {
         assertIsFocusWithin(node3);
     }
 
+    @Test public void testFocusWithinBitsAreSetAndClearedOnParentsOfNodeContainingMultipleFocuses() {
+        class N extends Group {
+            N(Node... children) { super(children); }
+            void _setFocused(boolean value) { setFocused(value); }
+        }
+
+        N node1, node2, node3, node4;
+
+        scene.setRoot(
+            node1 = new N(
+                node2 = new N()
+            ));
+
+        node3 = new N(
+            node4 = new N());
+
```

Comment:
ignore: I think this looks a bit inconsistent (in the first version not all parenthesis are closed on the last line, while they are in the second version).  Personally I favor this, similar to how curly braces are used):
```suggestion
        scene.setRoot(
            node1 = new N(
                node2 = new N()
            )
        );

        node3 = new N(
            node4 = new N()
        );

```

Reply:
Done.

Reply:
FYI: Using jfx21, I have observed a case where the count becomes negative. If I can create a minimal example, I will file a bug report.  However it's happening in a complex application, so it make take time for me to reduce it down to something I can reasonably provide.